service/app/src/app/game/[gameId]/setup/page.tsx

```diff
@@ -1,5 +1,159 @@
+    return "이미 사용중인 팀명이에요."
+  }
+  return ""
+}
+
+export default function GameSetupPage() {
+  const params = useParams()
+  const router = useRouter()
+
+  const { teams, addTeam, removeTeam, updateTeamName, setGameStatus } =
+    useGameStore(
+      useShallow((state) => ({
+        teams: state.teams,
+        addTeam: state.addTeam,
+        removeTeam: state.removeTeam,
+        updateTeamName: state.updateTeamName,
+        setGameStatus: state.setGameStatus,
+      })),
+    )
+
+  //팀에 대한 유효성 검사 - 전역 store에 추가하지 않고 관리
+  const teamErrors = useMemo(
+    () =>
+      teams.reduce(
+        (errors, team) => {
+          const error = validateTeamName(team.id, team.name, teams)
+          if (error) {
+            errors[team.id] = error
+          }
+          return errors
+        },
+        {} as { [teamId: string]: string },
+      ),
+    [teams],
+  )
+
+  const handleUpdateTeamName = (teamId: string, name: string) => {
+    if (name.length > MAX_TEAM_NAME_LENGTH) {
+      return
+    }
+    updateTeamName(teamId, name)
+  }
+
+  const handleAddTeam = () => {
+    if (teams.length >= MAX_TEAMS) {
+      return
+    }
+
+    const newId = uuidv4()
+    const newTeamName = `${String.fromCharCode(65 + teams.length)}팀`
+    addTeam({
+      id: newId,
+      name: newTeamName,
+      members: [],
+    })
+  }
+
+  const handleRemoveTeam = (teamId: string) => {
+    removeTeam(teamId)
+  }
+
+  const handleExitClick = () => {
+    openExitConfirmDialog({
+      onConfirm: () => router.push("/"),
+    })
+  }
+
+  return (
+    <div className="flex h-screen w-screen flex-col bg-background-primary">
+      <GameNavigation
+        onStart={() => {
+          setGameStatus("playing")
+          router.push(`/game/${params.gameId}/play`)
+        }}
+        onExit={handleExitClick}
+        canStart={Object.keys(teamErrors).length > 0}
+      />
+      <section className="flex flex-1 overflow-hidden">
+        <TeamSidebar teams={teams} />
+        <div className="flex h-full flex-1 flex-col items-center justify-center">
+          <div className="flex max-h-full min-w-[452px] flex-col items-center justify-start overflow-y-auto">
+            <InputComponents.Root
+              className="flex w-full flex-col gap-6"
+              onSubmit={(e) => {
+                e.preventDefault()
+                handleAddTeam()
+              }}
+            >
+              {teams.map((team, index) => (
+                <div key={team.id}>
+                  <InputComponents.Field
+                    name={`team-${team.id}-${index}`}
+                    type="reset"
+                    state={teamErrors[team.id] ? "error" : "default"}
+                  >
+                    <InputComponents.Control
+                      value={team.name}
+                      onChange={(value) => handleUpdateTeamName(team.id, value)}
+                      onReset={() => {
+                        if (teams.length > 2) {
```

Comment:
여기서 2도 상수로 빼면 좋을 것 같습니닷
```suggestion
const MIN_TEAMS = 2
```